app/frontend/wallet/shelley/shelley-transaction-planner.ts

```diff
@@ -286,57 +309,94 @@ export function computeTxPlan(
   // if we cannot create a change output with minimal ada we add it to the fee
   if (change.tokenBundle.length === 0 && change.coins < minimalChangeLovelace) {
```

Comment:
I see no logic for failure in case the change output contains tokens and the ADA is not enough to cover the minimum ADA per output - how is that covered?

Reply:
Check out the `validateTxPlan` function, it checks every output for its minimal Lovelace amount. Note that I intentionally moved all validations to the `validateTxPlan` function and left the `computeTxPlan` only the responsibility of computing a balanced plan.